packaging/selinux/microshift.te

```diff
@@ -4,11 +4,21 @@ type microshift_t;
 domain_type(microshift_t);
 
 gen_require(`
-    type container_runtime_t, var_lib_t, container_var_lib_t, container_runtime_tmpfs_t, container_t;
+    type container_runtime_t, var_lib_t, container_var_lib_t, container_runtime_tmpfs_t, container_t, bin_t;
     class file { open read };
 ')
 filetrans_pattern(container_runtime_t, var_lib_t, container_var_lib_t, dir, "kubelet")
 
+# When microshift creates backup folders in /var/lib/microshift-backups, the correct labels are applied
+# Note: filetrans_pattern rules states;
+#   Process running as `container_runtime_t` that creates a `dir` called `microshift-backups` 
+#   in a folder labeled `var_lib_t` get's the label `container_var_lib_t`
+filetrans_pattern(container_runtime_t, var_lib_t, container_var_lib_t, dir, "microshift-backups")
+
+# greenboot-healthcheck runs as an bin_t context, here we make sure the correct label is applied when 
+# greenboot creates the health.json file.
+filetrans_pattern(bin_t, var_lib_t, container_var_lib_t, dir, "microshift-backups")
```

Comment:
This is the same rule as the previous one, except applied to processes running in the bin_t domain - since health.json will inherit the label of it's parent, this is fine, but the comment is a little misleading, since it sounds like you intend to make a filetrans_pattern that applies to the health.json file specifically.

Reply:
Good point, updated. I'm annoyingly running into an issue here where green boot creates the `/var/lib/microshift-backups` folder but it seems it runs under a context I can't identify to get this `filetrans_pattern` to trigger. Luckily there is no difference in access since `container_t` types are not allowed to access `var_lib_t`